code/game/turfs/turf.dm

```diff
@@ -538,8 +564,41 @@
 			if(nutri_check.nutriment_factor >0)
 				M.reagents.remove_reagent(R.type, min(R.volume, 10))
 
-//Whatever happens after high temperature fire dies out or thermite reaction works.
-//Should return new turf
+/turf/proc/set_damage_amount(temperature)
+	if(to_be_destroyed && !changing_turf)
+		damage_amount = clamp((temperature - heat_capacity)/10000, 1, 15)
+		turf_take_damage(damage_amount)
+		heat_capacity = max(heat_capacity - (damage_amount * 10), 0)
+	else
+		to_be_destroyed = FALSE
+		max_fire_temperature_sustained = 0
+
+/turf/proc/turf_take_damage(damage_amount)
+	if(QDELETED(src))
+		stack_trace("[src] taking damage after deletion")
+		return
+	if(turf_heat_resistance <= 0)
+		return
+	turf_heat_resistance = max(turf_heat_resistance - damage_amount, 0)
+	update_overlays()
+	if(turf_heat_resistance <= 0)
+		Melt()
+
+/turf/update_overlays()
+	. = ..()
+	var/ratio = turf_heat_resistance / turf_max_heat_resistance
+	ratio = CEILING(ratio*4, 1) * 25
+	if(ratio >= 75)
+		return
+	if(ratio < 25)
+		overlays +=  image('icons/turf/walls.dmi', icon_state = "melt_25")
+	else if(ratio < 50)
+		overlays +=  image('icons/turf/walls.dmi', icon_state = "melt_50")
+	else if(ratio < 75)
+		overlays +=  image('icons/turf/walls.dmi', icon_state = "melt_75")
+	return
```

Comment:
Do a `switch` and don't use a `return` at the end of the proc, that's implied.

Reply:
okok

Reply:
how do you even use switch here, this if usage is perfectly fine

Reply:
```cpp
switch(ratio)
	if(-INFINITY to 24)
		stuff
	if(24to 49)
		stuff
	if(49 to 74)
		stuff
```